trinity/main.py

```diff
@@ -168,6 +164,21 @@ def main() -> None:
                 "inside the XDG_TRINITY_ROOT path"
             )
 
+    # Verify that the database engine that trinity is configured to use matches
+    # the existing on-disk engine.
+    if trinity_config.db_backend != trinity_config.on_disk_database_engine:
```

Comment:
Since both `db_backend` and `database_engine` are storing those constants in `{'rocks', 'level'}`, can we unify the names a bit? I think I like engine better, because `backend` implies to me that I'm going to get a `LevelDB` or `LevelDB()` back instead of `'level'`.

So maybe `config.db_engine` and `config.on_disk_db_engine`?